src/runtime/long.cpp

```diff
@@ -463,7 +467,14 @@ extern "C" PyObject* PyLong_FromSize_t(size_t ival) noexcept {
 #undef IS_LITTLE_ENDIAN
 
 extern "C" double _PyLong_Frexp(PyLongObject* a, Py_ssize_t* e) noexcept {
-    Py_FatalError("unimplemented");
+    BoxedLong* v = (BoxedLong*)a;
+    double result = mpz_get_d_2exp(e, v->n);
+    if (*e > INT_MAX) {
```

Comment:
Any reason to use INT_MAX here instead of PY_SSIZE_T_MAX?

If we're checking overflow against PY_SSIZE_T_MAX, I'm not sure we can use the result of mpz_get_d_2exp to do that; not really sure what gmp would do in that situation.  I think this only matters on 32-bit systems (otherwise the long object itself would require 2^60 bytes of memory to overflow here), so maybe we could replace this block with a `static_assert(sizeof(Py_ssize_t) == 8, "need to add overflow checking")`?